packages/next/src/client/components/router-reducer/get-prefetch-cache-entry-status.ts

```diff
@@ -0,0 +1,37 @@
+import { PrefetchCacheEntry } from './router-reducer-types'
+
+const FIVE_MINUTES = 5 * 60 * 1000
+const THIRTY_SECONDS = 5 * 1000
+
+export enum PrefetchCacheEntryStatus {
+  'reusable',
+  'expired',
+  'stale',
+}
+
+export function getPrefetchEntryCacheStatus({
+  kind,
+  prefetchTime,
+  lastUsedTime,
+}: PrefetchCacheEntry): PrefetchCacheEntryStatus {
+  // if the cache entry was prefetched or read less than 30s ago, then we want to re-use it
+  if (Date.now() < (lastUsedTime ?? prefetchTime) + THIRTY_SECONDS) {
```

Comment:
Every time the cache the user navigates, it updates the `lastUsedTime` right ? 
That means that if the user navigates to the page in multiple successions in a short time, the page will never be refetched ?

Reply:
Yeah, any concerns on that?

Reply:
The only concern i have is about stale data between navigations, especially for pages that are supposed to be dynamic. The assumption on dynamic pages is that they are rendered on every request including navigations.

I suppose that this is an optimization to not hit the server every often, so i have a just a question is it possible to make the caching logic dependant on the `revalidate` parameter on the page ? 

something like this : 
- if no `revalidate` constant is exported and the page is marked as dynamic consider revalidate as 0 
- if `revalidate` is exported use that configuration to mark the prefetch as stale/expired

Reply:
Keep in mind that the 30s timeout is for when you're visiting the same exact link. This scenario in particular is a heuristic for when you just want to quickly go back and forth.

For the client side revalidation, yes I'm thinking about something like that yeah. Not revalidate though, as revalidate is for the server.

Reply:
Ah my bad, for back & forth navigations this is pretty cool. 
Thank you for taking your time to answer have a good day.